### ISSUE TITLE
Intercept Altitudes

### ISSUE THREAD

**Issue by mooneybravo** [zlsa/atc#625]
Aircraft that are intercepting below the glideslope are penalized for high intercept angles. Realistically aircraft should be entering ILS approaches significantly below the glideslope to allow the pilot to fly the correct course before beginning descent. 

"The vector shall also enable the aircraft to be established on the ILS localizer course in level flight for at
least 2NM (3.7km) prior to intercepting the ILS glide path." - https://www.ivao.aero/training/documentation/books/APC_parallel_runways.pdf

Therefore controller should be penalised for aircraft entering high or too far right/left but not for aircraft that are low.

**Comment by erikquinn** [comment]
@mooneybravo really? I double checked the code and don't believe they should be getting penalized for high intercept angles in relation to their altitude, the two seem to be unrelated in how the logic is implemented and shouldn't affect it. Relevant section in `aircraft.js:2433`.

As for the latter, unless there is a similar mandate in the 7110.65 (I'm not aware of one), no penalties will be implemented for that. IVAO/VATSIM have plenty of policies that are not 100% in alignment with real world practice, and it's the real-world practice that we're trying to simulate.

**Comment by eliuuk** [comment]
There is no policy I believe about intercept altitudes, but what could be 
implemented is the maximum altitude the glidescope can be caught at, which 
is seen on Navigraph ILS charts.

**Comment by mooneybravo** [comment]
I thought (and was reasonably certain) I was observing this behavior. Just tried to replicate by vectoring an aircraft to the glide slope at 1000ft 20 miles from the runway and it established fine. Will see if I can replicate what I was seeing again, but basically I was observing that aircraft establishing on glidepath but not descending were being consistently penalised when intercept angle was 20° off nominal runway heading. 

As for the information on IVAO this was just showing that ideally aircraft would be entering below glideslope as I believe would be best practice if not actually mandated in the real world. A common real world ATC transmission might include 'maintain 3000 until established' or similar.

**Comment by erikquinn** [comment]
Gotcha. You're right, it's definitely best practice to bring ac to the localizer well below the GS, and that is the phraseology indeed. I just don't personally think I'd want to add or subtract any points for that. Now something like whether or not they intercept a sufficient distance out from the approach gate (outside the FAF), THAT, I would support a penalty for. We'd just have to add in the location of all final approach fixes for each runway of all airports, and I don't feel like doing that! haha :smiley:

**Comment by mooneybravo** [comment]
There is definitely some funny behavior with intercept angle penalisation. I was just penalised for this: 
![intercept](https://cloud.githubusercontent.com/assets/20625108/17465685/b43c0588-5cf6-11e6-9bde-c9a5334c94a0.JPG)
The trail shows direction assigned through the turn command before issuing clearance for ILS approach.

**Comment by mooneybravo** [comment]
And... I think I've worked it out. 
![22](https://cloud.githubusercontent.com/assets/20625108/17465776/43abeaca-5cf8-11e6-885a-19264cda4c67.JPG)
![intercept](https://cloud.githubusercontent.com/assets/20625108/17465777/43ad0e3c-5cf8-11e6-8644-3c4112075d25.JPG)

Vectoring and then using a fix along the ILS path triggers message because the last vector given to the aircraft may be significantly different to runway heading. Am I right that the logic is somehow using the heading command rather than the actual angle of interception? 

If that is the case, why does flying aircraft fix-to-fix behave as expected regarding the actual angle of interception?

**Comment by erikquinn** [comment]
@mooneybravo Okay, yeah, that would be true, because it checks the runway's "nominal heading" (eg 270 for Runway "27L") and compares it against the heading assigned in the active "waypoint".

```
            if (!this.projected && (abs(angle_offset(this.fms.currentWaypoint().heading,
                  radians(parseInt(this.rwy_arr.substr(0,2))*10))) > radians(30))) {
              ui_log(true, this.getRadioCallsign() +
                      " approach course intercept angle was greater than 30 degrees");
              prop.game.score.violation += 1;
            }
```

Should probably be changed to accommodate sending a/c direct to a fix to join an approach.

As for your last question, it's simply because there's no rules requiring ATC to do anything specific with the routes the aircraft fly, only to give a 30-degrees-or-less angle when vectoring them to intercept the ILS's final approach course.

Side note: My understanding of the rules is that in the instance where we send an aircraft direct to a fix that is not a designated IAF on the approach plate, the same rules for interception apply under FAA JO 7110.65, 5-9-2-a, just as if we had vectored them onto the final. However, if we clear them to an IAF, then ATC is not expected to provide any sort of directionality, and the presence/absence of a procedure turn would be dictated by the approach plate. In the former, we need to do the 30-degree intercept check, and in the latter, we would not need to. To simplify the implementation (since it'll be nightmarish to track all the IAFs for all approaches), for now, we can just make the 30-degree check happen both on vectors to final and fixes to final. If/when GPS approaches are implemented (espeically the T-shaped ones), we'll have to revisit this, because 90>30. :wink:

**Comment by erikquinn** [comment]
Resolved by #702.
